specification/attributes/discount_handling.md

```diff
@@ -0,0 +1,49 @@
+
+While providers may use different terms to describe discounts, FOCUS identifies a discount as being a reduced price applied directly to a row. Any price or cost reductions that are awarded after the fact are identified as a "Credit" Charge Subcategory. One example might be when a provider offers a reduced rate after passing a certain threshold of usage or spend.
+
+All rows defined in FOCUS MUST follow the discount handling requirements listed below.
+
+## Attribute ID
+
+DiscountHandling
+
+## Attribute name
+
+Discount Handling
+
+## Description
+
+Indicates how to include and apply discounts to usage charges or rows.
+
+## Requirements
+
+* All applicable discounts SHOULD be applied to each row they pertain to and SHOULD NOT be negated in a separate row.
+* All discounts applied to a row MUST apply to the entire charge.
+  * If a discount only applies to a subset of the charge, then that portion of the charge MUST be split into a separate row. One with the discount and one without.
+  * If multiple discounts apply, there MUST be separate rows for each unique combination of applied discounts or undiscounted amount.
+  * Each discount MUST be identifiable using existing FOCUS columns.
+    * Discounts applied to rows from a commitment-based discount MUST include a CommitmentDiscountId.
+    * If a provider applies a discount that cannot be represented by a FOCUS column, they SHOULD include additional columns to identify the source of the discount.
+* Purchased discounts (e.g., commitment-based discounts) MUST specify CommitmentUtilization on the usage rows that received the reduced rates from the discount.
+* Purchased discounts (e.g., commitment-based discounts) MUST be amortized.
+  * The BilledCost MUST be 0 for any row where the commitment covers the entire cost for the charge period.
+  * The EffectiveCost MUST be the portion of the amortized purchase cost that applies to this row.
+  * CommitmentUtilization MUST be "Used" for rows that received a reduced price from that commitment.
+  * If a commitment is not fully utilized, the provider MUST include a row that represents the unused portion of the commitment for that charge period. Charge Subcategory MUST be "Commitment Not Used".
+  * The sum of the EffectiveCost for all "Commitment Used" and "Commitment Not Used" rows for each ChargeSubcategory MUST be the same as the BilledCost of the commitment-based discount purchase.
```

Comment:
I think some of the confusion is coming from lines 36-40 not at low-enough level to indicate: 
> how to include and apply discounts to usage charges or rows


```suggestion
* Prepaid portions of purchased discounts (e.g., commitment-based discounts) MUST be amortized.
  * The BilledCost MUST be 0 for any row where ChargeType is "Usage" and the commitment applies to the entire charge for the charge period.
  * The EffectiveCost MUST include the portion of the amortized purchase cost proportional to the Pricing Unit and the time granularity of the charge, for any row that received a reduced price from a commitment.
  * ChargeSubcategory MUST be "Used Commitment" for rows that received a reduced price from that commitment.
  * The provider MUST include a row that represents unused portions of a commitment based on the time granularity of the dataset. ChargeSubcategory MUST be "Unused Commitment" for these rows.
```

I removed my draft of line 40 because ChargeSubCategory does not support "Purchase" yet.
>* The sum of BilledCost for all rows where ChargeSubcategory is "Used Commitment" or "Unused Commitment" MUST equal the value committed by all commitments within the time granularity of the dataset.